### PR TITLE
WIP: Create moveit_grasps tutorial

### PR DESCRIPTION
**Pull request by henningkayser**
Here is a draft for the moveit_grasps tutorial.

At the current state this is basically the README converted to RST format.

### CODE REVIEW

**Comment by henningkayser** [comment]
@davetcoleman, @mlautman I created this PR here so that we can complete/review this before targeting ros-planning/moveit_tutorials. Thought it would make sense this way before the release.

File: doc/moveit_grasps/moveit_grasps_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,140 @@
+MoveIt! Grasps
+=======================
```
Comment (davetcoleman) [inline doc/moveit_grasps/moveit_grasps_tutorial.rst]: Make = same width as title (i think this is required?)

File: doc/moveit_grasps/moveit_grasps_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+MoveIt! Grasps is a grasp generator for objects such as blocks or cylinders for use with the MoveIt! pick and place pipeline.
```
Comment (davetcoleman) [inline doc/moveit_grasps/moveit_grasps_tutorial.rst]:
```suggestion
MoveIt! Grasps is a grasp generator for objects such as blocks or cylinders for use with the MoveIt! pick and place pipeline. It also contains its own components for filtering grasps based on reachability and approach/retreat motions.
```

File: doc/moveit_grasps/moveit_grasps_tutorial.rst [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+The grasp generation algorithm is based on shapes and kinematics and does not consider friction cones or other dynamics.
```
Comment (davetcoleman) [inline doc/moveit_grasps/moveit_grasps_tutorial.rst]:
```suggestion
The grasp generation algorithm is based on simple cuboid shapes and does not consider friction cones or other grasp dynamics.
```

File: doc/moveit_grasps/moveit_grasps_tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+Its current implementation takes as input a pose vector (postition and orientation) and generates a large number of potential grasp approaches and directions.
```
Comment (davetcoleman) [inline doc/moveit_grasps/moveit_grasps_tutorial.rst]:
```suggestion
Its current implementation takes as input a pose of a cuboid, and the cuboid's size, and generates a large number of potential grasp approaches and directions.
```

File: doc/moveit_grasps/moveit_grasps_tutorial.rst [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+Also included is a grasp filter for removing kinematically infeasible grasps via threaded IK solvers.
+
+.. image:: https://raw.githubusercontent.com/PickNikRobotics/moveit_grasps/kinetic-devel/resources/demo.png
+   :width: 500pt
```
Comment (davetcoleman) [inline doc/moveit_grasps/moveit_grasps_tutorial.rst]: I think we could make a better screenshot, this one is old

File: doc/moveit_grasps/moveit_grasps_tutorial.rst [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+Installing MoveIt! Grasps
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ubuntu Debian
+--------------
+
+   Note: this package has not been released yet
+::
+
+    sudo apt-get install ros-kinetic-moveit-grasps
```
Comment (davetcoleman) [inline doc/moveit_grasps/moveit_grasps_tutorial.rst]: remove this section

File: doc/moveit_grasps/moveit_grasps_tutorial.rst [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+
+Install From Source
+--------------------
```
Comment (davetcoleman) [inline doc/moveit_grasps/moveit_grasps_tutorial.rst]: adapt to use franka emika workspace created by Getting Started tutorial

File: doc/moveit_grasps/moveit_grasps_tutorial.rst [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+    catkin build
+
+Setup
+^^^^^^^^^^^^^^^^
+
+Robot-Agnostic Configuration
+----------------------------
+
+You will first need a configuration file that described your robot's end effector geometry.
+Currently an example format can be seen in this repository at `config_robot/baxter_grasp_data.yaml <https://github.com/PickNikRobotics/moveit_grasps/blob/kinetic-devel/config_robot/baxter_grasp_data.yaml>`_.
```
Comment (davetcoleman) [inline doc/moveit_grasps/moveit_grasps_tutorial.rst]: deadlink. its now a panda example

File: doc/moveit_grasps/moveit_grasps_tutorial.rst [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+* BLUE - pregrasp filtered by ik
+* CYAN - pregrasp filtered by collision
+* GREEN - valid
+
+Tested Robots
+-------------
+
+* UR5
+* Jaco2
+* `Baxter <https://github.com/davetcoleman/baxter_cpp>`_
```
Comment (davetcoleman) [inline doc/moveit_grasps/moveit_grasps_tutorial.rst]: remove baxter link (old link)

File: doc/moveit_grasps/moveit_grasps_tutorial.rst [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+* Panda
+
+Example Code
+------------
+
+The most current example for using MoveIt! Grasps is the ``grasp_pipeline_demo`` which can be found `here <https://github.com/PickNikRobotics//moveit_grasps/kinetic-devel/src/grasp_pipeline_demo.cpp>`_.
+
+There are other example implementations:
+
+* `baxter_pick_place <https://github.com/davetcoleman/baxter_cpp/tree/kinetic-devel/baxter_pick_place>`_
```
Comment (davetcoleman) [inline doc/moveit_grasps/moveit_grasps_tutorial.rst]: i'd remove these examples, they are out of date

File: doc/moveit_grasps/moveit_grasps_tutorial.rst (hunk 1 above)
Comment (mlautman) [reply doc/moveit_grasps/moveit_grasps_tutorial.rst]: Not required

Comment (davetcoleman) [reply doc/moveit_grasps/moveit_grasps_tutorial.rst]: ack

**Pull request closed without merging**